gtk/PrefsDialog.cc

```diff
@@ -909,25 +911,48 @@ NetworkPage::~NetworkPage()
     portTag_.disconnect();
 }
 
-void NetworkPage::onPortTested(bool isOpen)
+void NetworkPage::onPortTested(std::string_view ip_protocol, bool is_open)
 {
-    portLabel_->set_markup(fmt::format(
-        isOpen ? _("Port is {markup_begin}open{markup_end}") : _("Port is {markup_begin}closed{markup_end}"),
-        fmt::arg("markup_begin", "<b>"),
-        fmt::arg("markup_end", "</b>")));
-    portButton_->set_sensitive(true);
-    portSpin_->set_sensitive(true);
+    static bool ipv4_complete = false;
+    static bool ipv6_complete = false;
+
+    if (ip_protocol == "ipv4")
```

Comment:
I'm okay with having strings in `Session` but passing them (and not e.g. enums) further up to `PrefsDialog` doesn't feel right; at least declare them as constants to not duplicate the literals.

Reply:
fd8dd0a GTK
a5aa955 Qt